mobile/Screen/MapFilter.js

```diff
@@ -0,0 +1,206 @@
+import React, { useState } from 'react';
+import {
+  View, TouchableOpacity, ScrollView, Text, StyleSheet,
+} from 'react-native';
+import PropTypes from 'prop-types';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    paddingHorizontal: 20,
+  },
+  text: {
+    fontSize: 16,
+    marginTop: 10,
+    marginLeft: 10,
+    color: 'grey',
+  },
+  header: {
+    color: 'dimgrey',
+    fontSize: 16,
+    marginTop: 25,
+    marginBottom: 10,
+    fontWeight: 'bold',
+    marginLeft: 10,
+  },
+  applyButton: {
+    marginTop: 20,
+    marginBottom: 50,
+    width: '95%',
+    borderRadius: 4,
+    height: 45,
+    backgroundColor: '#424288',
+    borderWidth: 2,
+    borderColor: '#424288',
+    justifyContent: 'center',
+    alignItems: 'center',
+    margin: 7,
+  },
+});
+
+function MapFilter({ route, navigation }) {
+  const {
+    categories, setCategories,
+  } = route.params;
+
+  const eventCategories = [
+    {
+      title: 'visual art',
+      length: 10,
```

Comment:
Instead of keeping a field of the string length hardcoded, can we use `title.length` and calculate the length every time we need it?

Reply:
additionally, instead of keeping a list of objects all with one `title` field, we can just make this a list of strings (i.e., a list of all the titles). This will require some code to be reworked.